src/main.elm

```diff
@@ -215,9 +202,16 @@ fromUrl url =
     String.dropLeft 2 url
 
 
-urlParser : Navigation.Parser String
-urlParser =
-    Navigation.makeParser (fromUrl << .hash)
+urlParser : Location -> Msg
```

Comment:
This seems a lot simpler on first blush... Interested to know what you think about it. I'll ask in channel.